doc/plugin-criteria.md

```diff
@@ -0,0 +1,129 @@
+Krew plugin criteria
+===
+
+Rationale
+---
+
+Krew is a `kubectl` plugin manager.
+
+The centralized Krew plugin index (`sigs.k8s.io/krew-index` repository) holds plugins that are applicable to a broad set of kubectl users.
+
+A plugin in the Krew plugin index must add new functionality or enhance the existing functionality in `kubectl`, or encapsulate a common operator/developer workflow for higher productivity.
+
+A plugin should be classified in the following categories:
+
+- Purpose
+- Category
+- Usability
+
+Purpose
+---
+
+This task should be classified in one of the categories:
+
+<table>
+    <thead>
+    <tr>
+        <th>#</th>
+        <th>Classification</th>
+        <th>Recommendation</th>
+    </tr>
+    </thead>
+    <tbody>
+    <tr>
+        <td>1.</td>
+        <td>The task is already solved by kubectl.</td>
+        <td>
+          Mere command wrappers are not acceptable, as those are better coveredby shell aliases.
+          However, convenience and productivity improvements are welcome. Examples:
+           <ul>
+             <li><code>konfig</code> is a convencience script to aid with merging of kubeconfig files, similar to <code>kubectl config view</code>.</li>
+             <li><code>change-ns</code> is a utility easily switch between namespaces, similar to <code>kubectl config set-context --curent -n</code></li>
+             <li><code>view-secret</code> retrieves and decodes secrets, similar to <code>kubectl get secret -o jsonpath=... | base64 -d</code></li>
+           </ul>
+        </td>
+    </tr>
+    <tr>
+        <td>2.</td>
+        <td>
+          The task provides a different solution to a problem which is also solved by kubectl.
+          However, the task would be impossible or very hard to reproduce with kubectl alone.
+        </td>
+        <td>
+          In general welcome unless it overlaps significantly with other plugins or has a bad usability.
+          Examples:
+          <ul>
+            <li><code>get-all</code> plugin is a replacement for <code>kubectl get all</code> command which actually fetches only a subset of API resource types.</li>
+            <li><code>access-matrix</code> gives an overview of authorities, whereas <code>kubectl auth can-i</code> only applies to single resource/verb pairs</li>
+            <li><code>tail</code> plugin allows tailing logs from pods from a label selector, a namespace or all namespaces at once, whereas <code>kubectl logs</code> command has more limited functionality.</li>
+            <li><code>mtail</code> plugin allows tailing logs from pods from a label selector, similar to <code>kubectl logs -l</code> but labels log lines.</li>
+          </ul>
+        </td>
+    </tr>
+    <tr>
+        <td>3.</td>
+        <td>The task addresses a completely different problem than any of the kubectl subcommands, but is clearly focused on k8s.</td>
```

Comment:
I think an indicator of this is: 

> Most users will use this tool as a kubectl plugin.

If this is the case, then it's probably more suitable. e.g. in case of `helm` vs `kubectl helm`, there's no need to make it a plugin, because it doesn't add anything to kubectl. 

Worth adding.

Reply:
I've added that to the recommendation column. Was that what you intended?